src/beamlime/constructors/inspectors.py

```diff
@@ -37,11 +37,25 @@ def validate_annotation(annotation) -> Literal[True]:
 
 
 Product = TypeVar("Product")
-ProductType = Type[Product]
+_NewT = NewType("_NewT", int)
 
 
-def extract_underlying_product_type(product_type: ProductType) -> ProductType:
-    if hasattr(product_type, "__supertype__"):  # NewType
+class NewTypeMeta(type):
+    def __instancecheck__(cls, __instance: Any) -> bool:
```

Comment:
During the runtime, `NewType` returns a dynamically created function, that is same as `lambda x: x`,
and typing module doesn't provide type check of `NewType` object as far as I know.